Android/app/src/main/java/tie/hackathon/travelguide/destinations/description/FinalCityInfo.java

```diff
@@ -182,57 +165,60 @@ public void onClick(View v) {
+        super.onStart();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+    }
 
+    @Override
+    public void showProgress() {
         dialog = new MaterialDialog.Builder(FinalCityInfo.this)
                 .title(R.string.app_name)
                 .content("Please wait...")
                 .progress(true, 0)
                 .show();
+    }
 
-        // to fetch city names
-        String uri = Constants.apilink + "city/info.php?id=" + id;
-        Log.e("executing", uri + " ");
-
-        //Set up client
-        OkHttpClient client = new OkHttpClient();
-        //Execute request
-        Request request = new Request.Builder()
-                .url(uri)
-                .build();
-        //Setup callback
-        client.newCall(request).enqueue(new Callback() {
-            @Override
-            public void onFailure(Call call, IOException e) {
-                Log.e("Request Failed", "Message : " + e.getMessage());
-            }
-
-            @Override
-            public void onResponse(Call call, final Response response) throws IOException {
-
-                final String res = response.body().string();
-                mHandler.post(() -> {
-                    try {
-                        JSONObject ob = new JSONObject(res);
-                        description = ob.getString("description");
-                        des.setText(description);
-                        Picasso.with(FinalCityInfo.this).load(ob.getJSONObject("weather").getString("icon")).into(ico);
-                        temp.setText(ob.getJSONObject("weather").getString("temprature") + (char) 0x00B0 + " C ");
-                        humidity.setText("Humidity : " + ob.getJSONObject("weather").getString("humidity"));
-                        weatherinfo.setText(ob.getJSONObject("weather").getString("description"));
-                        lat = ob.getString("lat");
-                        lon = ob.getString("lng");
-
-                    } catch (JSONException e) {
-                        e.printStackTrace();
-                        Log.e("EXCEPTION : ", e.getMessage() + " ");
-                    }
-                    dialog.dismiss();
-                });
-            }
+    @Override
+    public void dismissProgress() {
+        dialog.dismiss();
+    }
+
+    @Override
+    public void parseResult(String description, String iconUrl, String temp, String humidity, String weatherInfo,
+                            String lat, String lon) {
+        mHandler.post(() -> {
+            des.setText(description);
+            Picasso.with(FinalCityInfo.this).load(iconUrl).into(ico);
+            this.temp.setText(temp + (char) 0x00B0 + " C ");
+            this.humidity.setText("Humidity : " + humidity);
+            weatherinfo.setText(weatherInfo);
+            this.lat = lat;
+            this.lon = lon;
         });
     }
+
+    private void fireIntent(Intent intent, String type) {
+        intent.putExtra("id_", id);
```

Comment:
Won't it be better, if, instead of passing keys("id_", "lat_",....) as string, we define them somewhere in the constants file, and implement it here, and use those variables.